src/deadline/nuke_submitter/ui/components/scene_settings_tab.py

```diff
@@ -48,31 +51,147 @@ def _build_ui(self):
+        lyt.addWidget(self.write_node_box, 0, 1, 1, -1)
 
         self.views_box = QComboBox(self)
         self.views_box.addItem("All Views", "")
         for view in sorted(nuke.views()):
             self.views_box.addItem(view, view)
         lyt.addWidget(QLabel("Views"), 1, 0)
-        lyt.addWidget(self.views_box, 1, 1)
+        lyt.addWidget(self.views_box, 1, 1, 1, -1)
 
         self.frame_override_chck = QCheckBox("Override Frame Range", self)
         self.frame_override_txt = QLineEdit(self)
         lyt.addWidget(self.frame_override_chck, 2, 0)
-        lyt.addWidget(self.frame_override_txt, 2, 1)
+        lyt.addWidget(self.frame_override_txt, 2, 1, 1, -1)
         self.frame_override_chck.stateChanged.connect(self.activate_frame_override_changed)
 
         self.proxy_mode_check = QCheckBox("Use Proxy Mode", self)
         lyt.addWidget(self.proxy_mode_check, 3, 0)
 
+        self.timeout_checkbox = QCheckBox("Use Timeouts", self)
+        self.timeout_checkbox.setChecked(True)
+        self.timeout_checkbox.clicked.connect(self.activate_timeout_changed)
+        self.timeout_checkbox.setToolTip(
+            "Set a maximum duration for actions from this job. See AWS Deadline Cloud documentation to learn more"
+        )
+        lyt.addWidget(self.timeout_checkbox, 4, 0)
+        self.timeouts_subtext = QLabel("Set a maximum duration for actions from this job")
+        self.timeouts_subtext.setStyleSheet("font-style: italic")
+        lyt.addWidget(self.timeouts_subtext, 4, 1, 1, -1)
+
+        self.timeouts_box = QGroupBox()
+        timeouts_lyt = QGridLayout(self.timeouts_box)
+        lyt.addWidget(self.timeouts_box, 5, 0, 1, -1)
+
+        def create_timeout_row(label, tooltip, row):
+            qlabel = QLabel(label)
+            qlabel.setToolTip(tooltip)
+            timeouts_lyt.addWidget(qlabel, row, 0)
+
+            days_box = QSpinBox(self, minimum=0, maximum=365)
+            days_box.setSuffix(" days")
+            timeouts_lyt.addWidget(days_box, row, 1)
+
+            hours_box = QSpinBox(self, minimum=0, maximum=23)
+            hours_box.setSuffix(" hours")
+            timeouts_lyt.addWidget(hours_box, row, 2)
+
+            minutes_box = QSpinBox(self, minimum=0, maximum=59)
+            minutes_box.setSuffix(" minutes")
+            timeouts_lyt.addWidget(minutes_box, row, 3)
+
+            return qlabel, days_box, hours_box, minutes_box
+
+        def hookup_zero_callback(timeout_boxes: tuple[QLabel, QSpinBox, QSpinBox, QSpinBox]):
+            def indicate_is_valid_callback(value: int):
+                self.indicate_if_valid(timeout_boxes)
+
+            for timeout_box in timeout_boxes[1:]:
+                timeout_box.valueChanged.connect(indicate_is_valid_callback)
+
+        self.on_run_timeouts = create_timeout_row(
+            label="Render Timeout",
+            tooltip="Maximum duration of each action which performs a render. Default is 6 days.",
```

Comment:
Is there a way to tweak this text to clarify that it might be per frame? Maybe "Render task timeout" since the user will see the submission represented as a series of tasks in the monitor.

Reply:
Yeah, I agree - `Render Task Timeout` would be more clear. 

Updated this in the latest commit